### PR TITLE
Improvements for the CH4 subconfigure.m4

### PR DESCRIPTION
**Pull request by raffenet**

### CODE REVIEW

**Comment by raffenet** [comment]
test:jenkins/ch3/all
test:jenkins/ch4/ofi
test:jenkins/ch4/ucx

**Comment by raffenet** [comment]
@LenaO I dropped the `gethostname` patch in order to re-work is slightly. I'll ping you again when it's set to review.

**Comment by raffenet** [comment]
test:jenkins/ch3/tcp
test:jenkins/ch4/ofi
test:jenkins/ch4/ucx

**Comment by raffenet** [comment]
@LenaO I decided to punt on a more extensive change to our checks for gethostname until later. This PR is ready for review.

File: src/mpid/ch4/subconfigure.m4 [hunk 1]
```diff
@@ -390,6 +381,14 @@ else
     AC_MSG_NOTICE([Using a non-symmetric heap])
 fi
 
+AC_CONFIG_FILES([
+src/mpid/ch4/src/mpid_ch4_net_array.c
+src/mpid/ch4/include/netmodpre.h
+src/mpid/ch4/include/shmpre.h
+])
+])dnl end AM_COND_IF(BUILD_CH4,...)
```
Comment (raffenet) [inline src/mpid/ch4/subconfigure.m4]: @LenaO the diff makes it a little hard to see, but what I'm doing is effectively moving the end of the `AM_COND_IF(BUILD_CH4)` here so the `AC_CHECK_HEADERS` and `gl_FUNC_RANDOM_R` checks are inside it. There are 2 `BUILD_CH4` conditionals in this file. One sets some variables, the other runs checks. This is similar to how the ch3 subconfigure does it.

**Comment by LenaO** [comment]
@raffenet Good, I finally see what commit 3 is doing.
I'm going to merge it.

**Pull request merged**